source/adapter.js

```diff
@@ -1048,16 +1048,11 @@ if ( navigator.mozGetUserMedia ||
     };
 
     getUserMedia = function (constraints, successCallback, failureCallback) {
-      var cc = {
-        audio: false,
-        video: false
-      };
-      if (constraints.audio) {
-        cc.audio = constraintsToPlugin(constraints.audio);
-      }
-      if (constraints.video) {
-        cc.video = constraintsToPlugin(constraints.video);
-      }
+      var cc = {};
+      cc.audio = constraints.audio ?
+        constraintsToPlugin(constraints.audio) : false;
```

Comment:
I'm not sure if setting `"audio"` or `"video"` constraints to `false` always a requirement in the plugin. Something to note is that providing the following constraints:

```
{
      video: {
        height: {
          min: 123,
          max: 456,
          ideal: 342
        },
        width: {
          min: 523,
          max: 856,
          ideal: 742
        }
      }
    }
```

**Chrome output:**
```
{"video":{"optional":[{"minHeight":342},{"maxHeight":342},{"minWidth":742},{"maxWidth":742}],"mandatory":{"minHeight":123,"maxHeight":456,"minWidth":523,"maxWidth":856}}}
```

**Safari (plugin):**
```
{"audio":false,"video":{"optional":[{"minHeight":342},{"maxHeight":342},{"minWidth":742},{"maxWidth":742}],"mandatory":{"minHeight":123,"maxHeight":456,"minWidth":523,"maxWidth":856}}}
```

Additional notes is good to have a log on the output shimmed constraints like: `console.log("plugin: " + JSON.stringify(cc));` following Chrome and Firefox (less than version `38`) `getUserMedia`.